UVtools.Core/EmguCV/EmguContours.cs

```diff
@@ -208,5 +209,32 @@ public static double[] GetContoursArea(List<VectorOfVectorOfPoint> contours, boo
             
             return result;
         }
+
+        public static bool CheckContoursIntersect(VectorOfVectorOfPoint contour1, VectorOfVectorOfPoint contour2)
+        {
+            var contour1Rect = CvInvoke.BoundingRectangle(contour1[0]);
+            var contour2Rect = CvInvoke.BoundingRectangle(contour2[0]);
+
```

Comment:
Before the hard code, check if both bouding rectangles intersect, if not return false and not worth to continue with code.

Reply:
This has been addressed with a contour1Rect.IntersectsWith(contour2Rect) check and exiting if they do not.